x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/EsRelation.java

```diff
@@ -112,7 +112,10 @@ public boolean equals(Object obj) {
         }
 
         EsRelation other = (EsRelation) obj;
-        return Objects.equals(index, other.index) && indexMode == other.indexMode() && frozen == other.frozen;
+        return Objects.equals(index, other.index)
+            && indexMode == other.indexMode()
+            && frozen == other.frozen
+            && Objects.equals(attrs, other.attrs);
```

Comment:
I think it's quite important; this can mask test failures (it did), and it can also prevent optimizations from taking effect if all they do is change an `EsRelation`'s attributes (like for union types).